### PR TITLE
Add LO link and void API

### PR DESCRIPTION
**Pull request by gdethier**
* Adds LO API for link management (add, delete, publish)
* Adds LO void API (with and without replacement)

logion-network/logion-internal#941

### CODE REVIEW

File: packages/client/src/Loc.ts [hunk 1]
```diff
@@ -914,6 +947,39 @@ export class LegalOfficerEditableRequestCommands {
...
+            link: {
+                id: parameters.target,
+                nature: hashString(link.nature),
+            },
+            signer: parameters.signer,
+            callback: parameters.callback,
+        });
+        return await this.request.refresh() as EditableRequest;
+    }
 }
```
Comment (benoitdevos) [inline packages/client/src/Loc.ts:983]: I don't think we need to define `publishLink()` at `EditableRequest` level. From what I can see, it is anyway overridden in `LegalOfficerOpenRequestCommands`, and it makes no sense to have it on a draft request.
Similarly, in user SDK, `publishMetadata` and `publishFile` only exist on `openLoc`.

**Review by benoitdevos — CHANGES_REQUESTED** [review]
I think one of the 2 provided `publishLink()` commands is useless.
Let's discuss if needed.
Otherwise, good job !

File: packages/client/src/Loc.ts (hunk 1 above)
Comment (gdethier) [reply packages/client/src/Loc.ts:983]: You're right, I forgot to remove the function from user API. Will do that now.

**Review by benoitdevos — APPROVED** [review]
:+1:

**Pull request merged**